str.go

```diff
@@ -0,0 +1,74 @@
+package merkletree
+
+import (
+	"github.com/coniks-sys/libmerkleprefixtree-go/crypto"
+	"github.com/coniks-sys/libmerkleprefixtree-go/internal"
+)
+
+// SignedTreeRoot represents a signed tree root, which is generated
+// at the beginning of every epoch.
+// Signed tree roots contain the current root node,
+// the current and previous counters, the hash of the
+// previous STR, and its signature.
+// STR should be final
+type SignedTreeRoot struct {
+	tree        *MerkleTree
+	counter     int64
+	prevCounter int64
+	prevStrHash []byte
+	sig         []byte
+	policies    Policies
+}
+
+func NewSTR(key crypto.SigningKey, policies Policies, m *MerkleTree, counter int64, prevHash []byte) *SignedTreeRoot {
+	str := &SignedTreeRoot{
+		tree:        m,
+		counter:     counter,
+		prevCounter: counter - 1,
+		prevStrHash: prevHash,
+		policies:    policies,
+	}
+	bytesPreSig := str.getSTRBytesForSig()
+	str.sig = crypto.Sign(key, bytesPreSig)
+	return str
+}
+
+func (str *SignedTreeRoot) getSTRBytesForSig() []byte {
+	var strBytes []byte
+	strBytes = append(strBytes, util.LongToBytes(str.counter)...)     // t - counter number
+	strBytes = append(strBytes, util.LongToBytes(str.prevCounter)...) // t_prev - previous counter number
+	strBytes = append(strBytes, str.tree.root.serialize()...)         // root
```

Comment:
Yeah, the root node is just an interior node (maybe we don't make this explicit in the paper? If not, then our bad). So root should also be ``h.child.0 || h.child.1``.

What we do in the reference implementation is just a vestige of an old root node format which contained many other fields (now in the STR). Sorry about this confusion! I think using ``treeHash`` in the STR makes perfect sense. And I should go back and fix this in the reference implementation, too.